GoogleForCreators/web-stories-wp
# Editor: Text sets can now be dragged and dropped onto page

## Summary

This PR adds drag-n-drop functionality to our text sets.

## Relevant Technical Choices

In order to be able to size the text set drag icon to its actual rendered size on the page (within the Editor), I had to create a context provider that wrapped both the `Library` and the `Canvas`.  I called it `Layout` and it stores the current `canvasPageSize` and provides an action to `setCanvasPageSize`.  I then updated the `Canvas` provider so it no longer holds `pageSize` locally, it routes its `pageSize` logic to the `Layout` provider.   This way elements in the `Library` and `Canvas` can both size themselves to the current canvas page size.  

And finally, using the `Layout` provider from within the `Library`, I was able to size the drag icon for the selected text set so it matches the exact size of its rendered self.

## User-facing changes

None.  Hidden behind flag.

## Testing Instructions

1.) Make sure "Enable text sets" is checked on the Experiments page.
2.) Open up a new/exisiting story in the editor.
3.) Click on the "T" in the libary panel to open up the Text tab.
4.) Use your mouse to drag and drop a text set onto the screen.  Note that the size and position of the dragged text set should match the dropped and rendered text set (however, if you drop the text set outside the boundaries of the page, it'll be adjusted to within the page on drop).

Fixes #4079 

## Screenshot

Drag and drop text set to page.  Notice the rendered text set's top-left corner aligns with the drag icon's top-left corner:
![better_drag_n_drop](https://user-images.githubusercontent.com/40646372/94202396-6efca780-fe72-11ea-88ec-3338843084b9.gif)

Drag and drop the text set outside the page, and we'll adjust it to fit onto the page nicely:
![boudary_protection](https://user-images.githubusercontent.com/40646372/94202404-72902e80-fe72-11ea-8686-0596c9136e1e.gif)
